synthesizer/snark/src/proving_key/mod.rs

```diff
@@ -91,3 +91,21 @@ impl<N: Network> Deref for ProvingKey<N> {
         &self.proving_key
     }
 }
+
+impl<N: Network> PartialEq for ProvingKey<N> {
+    fn eq(&self, other: &Self) -> bool {
+        self.deref() == other.deref()
+    }
+}
+
+impl<N: Network> Ord for ProvingKey<N> {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.deref().cmp(other.deref())
+    }
+}
+
+impl<N: Network> PartialOrd for ProvingKey<N> {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
```

Comment:
Why can't these be auto-derived?

Reply:
error[E0277]: the trait bound `N: Ord` is not satisfied
    --> synthesizer/process/src/trace/mod.rs:89:31
     |
89   |         self.transition_tasks.entry(proving_key).or_default().push(assignment);
     |                               ^^^^^ the trait `Ord` is not implemented for `N`
     |
     = note: required for `ProvingKey<N>` to implement `Ord`

Reply:
That's very odd, I'll give it a look tomorrow as well.

Reply:
I think this would do the job: https://github.com/AleoHQ/snarkVM/pull/2231